src/node.cc

```diff
@@ -1198,7 +1198,10 @@ Local<Value> MakeCallback(Environment* env,
   }
 
   if (ret.IsEmpty()) {
-    return Undefined(env->isolate());
+    if (callback_scope.in_makecallback())
```

Comment:
Would it be useful to add a comment here mentioning that `undefined` is returned here for backward compatibility?